File: src/dev/resources/tester_bolt.rb
require File.expand_path("storm", File.dirname(__FILE__))

class TesterBolt < Storm::Bolt
  def prepare(conf, context)
    emit ['bolt initializing']
  end

  def process(tuple)
    word = tuple.values[0]
    if (rand < 0.75)
      emit [word + "lalala"], :anchor => tuple
      ack tuple
    else
      log(word + ' randomly skipped!')
    end
  end
end

TesterBolt.new.run

File: src/dev/resources/tester_bolt.py
# This Python file uses the following encoding: utf-8

import storm
from random import random

class TesterBolt(storm.Bolt):
    def initialize(self, conf, context):
        storm.emit(['bolt initializing'])

    def process(self, tup):
        word = tup.values[0];
        if (random() < 0.75):
            storm.emit([word + 'lalala'], anchors=[tup])
            storm.ack(tup)
        else:
            storm.log(word + ' randomly skipped!')

TesterBolt().run()
